drivers/flash/flash_stm32_ospi.c

```diff
@@ -519,14 +519,40 @@ static bool ospi_address_is_valid(const struct device *dev, off_t addr,
 	return (addr >= 0) && ((uint64_t)addr + (uint64_t)size <= flash_size);
 }
 
+static int stm32_ospi_wait_auto_polling(struct flash_stm32_ospi_data *dev_data,
+		OSPI_AutoPollingTypeDef *s_config, uint32_t timeout_ms)
+{
+	/* If interrupt is enabled, no need for busy waiting to check the poll result
+	 * like HAL_OSPI_AutoPolling() does.
+	 */
+	if (irq_is_enabled(DT_IRQN(STM32_OSPI_NODE))) {
```

Comment:
I don't see cases where irq could be disabled in current driver implementation. Do you have something in mind ?

Reply:
The branch was left there to handle the polling in driver initialization before the interrupt is enabled.

But I think you're correct that there's no such case where interrupts are disabled but poll is ongoing. So indeed the else branch could be removed for now at least. I'll update the PR.